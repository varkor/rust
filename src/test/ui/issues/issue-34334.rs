fn main () {
    let sr: Vec<(u32, _, _) = vec![];
    //~^ ERROR expected one of `,` or `>`, found `=`
    //~| ERROR expected value, found struct `Vec`
    //~| ERROR expected value, found builtin type `u32`
    //~| ERROR mismatched types
    //~| ERROR invalid left-hand side of assignment
    //~| ERROR expected expression, found reserved identifier `_`
    //~| ERROR expected expression, found reserved identifier `_`
    let sr2: Vec<(u32, _, _)> = sr.iter().map(|(faction, th_sender, th_receiver)| {}).collect();
    //~^ ERROR a value of type `Vec<(u32, _, _)>` cannot be built
}
