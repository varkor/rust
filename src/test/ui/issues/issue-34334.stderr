error: expected one of `,` or `>`, found `=`
  --> $DIR/issue-34334.rs:2:29
   |
LL |     let sr: Vec<(u32, _, _) = vec![];
   |         --                  ^ expected one of `,` or `>`
   |         |
   |         while parsing the type for `sr`

error[E0423]: expected value, found struct `Vec`
  --> $DIR/issue-34334.rs:2:13
   |
LL |     let sr: Vec<(u32, _, _) = vec![];
   |             ^^^ did you mean `Vec { /* fields */ }`?

error[E0423]: expected value, found builtin type `u32`
  --> $DIR/issue-34334.rs:2:18
   |
LL |     let sr: Vec<(u32, _, _) = vec![];
   |                  ^^^ not a value

error: expected expression, found reserved identifier `_`
  --> $DIR/issue-34334.rs:2:23
   |
LL |     let sr: Vec<(u32, _, _) = vec![];
   |                       ^ expected expression

error: expected expression, found reserved identifier `_`
  --> $DIR/issue-34334.rs:2:26
   |
LL |     let sr: Vec<(u32, _, _) = vec![];
   |                          ^ expected expression

error[E0308]: mismatched types
  --> $DIR/issue-34334.rs:2:31
   |
LL |     let sr: Vec<(u32, _, _) = vec![];
   |                               ^^^^^^ expected `bool`, found struct `std::vec::Vec`
   |
   = note: expected type `bool`
            found struct `std::vec::Vec<_>`
   = note: this error originates in a macro (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0070]: invalid left-hand side of assignment
  --> $DIR/issue-34334.rs:2:29
   |
LL |     let sr: Vec<(u32, _, _) = vec![];
   |             --------------- ^
   |             |
   |             cannot assign to this expression

error[E0599]: no method named `iter` found for unit type `()` in the current scope
  --> $DIR/issue-34334.rs:10:36
   |
LL |     let sr2: Vec<(u32, _, _)> = sr.iter().map(|(faction, th_sender, th_receiver)| {}).collect();
   |                                                                                       ^^^^^^^ value of type `Vec<(u32, _, _)>` cannot be built from `std::iter::Iterator<Item=()>`
   |
   = help: the trait `FromIterator<()>` is not implemented for `Vec<(u32, _, _)>`

error: aborting due to 8 previous errors

For more information about this error, try `rustc --explain E0277`.
